### ISSUE TITLE
nxos_feature is broken with CLI transport (show feature command doesn't generate JSON output)

### ISSUE THREAD

**Issue by ipspace** [ansible/ansible-modules-core#4402]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

nxos_feature
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /vagrant/Ansible/Networking/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[defaults]
inventory=./hosts
gathering=explicit
transport=local
```
##### OS / ENVIRONMENT

OS: Ubuntu
Remote: NXOS 7.2(0)D1(1)
##### SUMMARY

nxos_feature executes "show feature | json" on Nexus OS. This command does not return JSON printout, resulting in task failure
##### STEPS TO REPRODUCE

Execute the following playbook on a Nexus OS

```

---
- hosts: s1.lab.local
  tasks:
  - nxos_feature:
      feature:  "{{item}}"
      state:    enabled
      provider: "{{cli}}"
    with_items: [ ospf ]
```
##### ACTUAL RESULTS

```
TASK [nxos_feature] ************************************************************
task path: /vagrant/Ansible/Networking/nexus-feature.yml:4
Using module file /usr/local/lib/python2.7/dist-packages/ansible-2.2.0-py2.7.egg/ansible/modules/core/network/nxos/nxos_feature.py
<s1.lab.local> ESTABLISH LOCAL CONNECTION FOR USER: vagrant
<s1.lab.local> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470980412.12-154171327302347 `" && echo ansible-tmp-1470980412.12-154171327302347="` echo $HOME/.ansible/tmp/ansible-tmp-1470980412.12-154171327302347 `" ) && sleep 0'
<s1.lab.local> PUT /tmp/tmpYiIS9g TO /home/vagrant/.ansible/tmp/ansible-tmp-1470980412.12-154171327302347/nxos_feature.py
<s1.lab.local> EXEC /bin/sh -c 'chmod -R u+x /home/vagrant/.ansible/tmp/ansible-tmp-1470980412.12-154171327302347/ && sleep 0'
<s1.lab.local> EXEC /bin/sh -c '/usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1470980412.12-154171327302347/nxos_feature.py; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1470980412.12-154171327302347/" > /dev/null 2>&1 && sleep 0'
failed: [s1.lab.local] (item=ospf) => {
    "command": "show feature | json",
    "failed": true,
    "invocation": {
        "module_args": {
            "feature": "ospf",
            "host": "s1.lab.local",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": null,
            "provider": {
                "host": "s1.lab.local",
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
                "transport": "cli",
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
            },
            "ssh_keyfile": null,
            "state": "enabled",
            "timeout": 10,
            "transport": "cli",
            "use_ssl": false,
            "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "validate_certs": true
        },
        "module_name": "nxos_feature"
    },
    "item": "ospf",
    "msg": "Command does not support JSON output"
}
```

**Comment by ansibot** [comment]
@privateip, @GGabriele, @gundalow, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ipspace** [comment]
It seems to be a Nexus OS software release/platform issue. I'd assume that **show feature** returns JSON object (when requested) on Nexus 9000, but it doesn't do that on Nexus 7000 (or at least not in Release 7.2). I can't possibly guess how hard it would be to add parsing of text printout, but it would be nice to have this module working on both platforms (and VIRL).

**Comment by gundalow** [comment]
@jedelman8 @GGabriele Could you please have a look at this, this seems like a valid 2.2 bug in your code.

**Comment by privateip** [comment]
This issue is believed to be fixed, please re-open the issue if you are still experiencing a problem.
